app/models/account/entry.rb

```diff
@@ -148,6 +148,27 @@ def search(params)
       query = query.where("account_entries.date >= ?", params[:start_date]) if params[:start_date].present?
       query = query.where("account_entries.date <= ?", params[:end_date]) if params[:end_date].present?
 
+      if params[:types].present?
+        query = query.where(marked_as_transfer: false) unless params[:types].include?("transfer")
+
+        if params[:types].include?("income") && !params[:types].include?("expense")
+          query = query.where("account_entries.amount < 0")
+        elsif params[:types].include?("expense") && !params[:types].include?("income")
+          query = query.where("account_entries.amount >= 0")
+        end
+      end
+
+      if params[:amount].present? && params[:amount_operator].present?
+        case params[:amount_operator]
+        when "equal"
+          query = query.where("ABS(ABS(account_entries.amount) - ?) <= 0.01", params[:amount].to_f.abs)
+        when "less"
+          query = query.where("ABS(account_entries.amount) < ?", params[:amount].to_f.abs)
+        when "greater"
+          query = query.where("ABS(account_entries.amount) > ?", params[:amount].to_f.abs)
+        end
+      end
```

Comment:
If we have to add more filters in the future, this is definitely a refactor candidate, but given this is the final set of filters for a while, no need right now.